File: frontend/src/features/transacoes/types/transacao.ts
export enum TipoTransacao {
  Despesa = 0,
  Receita = 1,
}

export interface Transacao {
  id: string;
  descricao: string;
  valor: number;
  tipo: TipoTransacao;
  categoriaId: string;
  pessoaId: string;
}

export interface CreateTransacaoDto {
  descricao: string;
  valor: number;
  tipo: TipoTransacao;
  categoriaId: string;
  pessoaId: string;
}

export const TipoTransacaoLabels: Record<TipoTransacao, string> = {
  [TipoTransacao.Despesa]: 'Despesa',
  [TipoTransacao.Receita]: 'Receita',
};

File: backend/ControleGastos.Models/Models/Transacao.cs
using System;
using ControleGastos.Models.Enums;
namespace ControleGastos.Models
{
    public class Transacao
    {
        public Guid Id { get; set; }
        public string Descricao { get; set; } = string.Empty;
        public decimal Valor { get; set; }
        public TipoTransacao Tipo { get; set; }
        public Guid CategoriaId { get; set; }
        public Guid PessoaId { get; set; }
    }
}
